packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -102,7 +104,96 @@ exports.sourceNodes = async (
     return
   }
 
-  const drupalFetchActivity = reporter.activityTimer(`Fetch data from Drupal`)
+  fastBuilds = fastBuilds || false
+  if (fastBuilds) {
+    let lastFetched = 0
+    if (
+      store.getState().status.plugins &&
+      store.getState().status.plugins[`gatsby-source-drupal`] &&
+      store.getState().status.plugins[`gatsby-source-drupal`].lastFetched
+    ) {
+      lastFetched = store.getState().status.plugins[`gatsby-source-drupal`]
+        .lastFetched
+    }
+
+    const drupalFetchIncrementalActivity = reporter.activityTimer(
+      `Fetch incremental changes from Drupal`
+    )
+    let requireFullRebuild = false
+
+    drupalFetchIncrementalActivity.start()
+
+    try {
+      // Hit fastbuilds endpoint with the lastFetched date.
+      const data = await axios.get(
+        `${baseUrl}/gatsby-fastbuilds/sync/${lastFetched}`,
+        {
+          auth: basicAuth,
+          headers,
+          params,
+        }
+      )
+
+      if (data.data.status === -1) {
+        reporter.info(`Unable to pull incremental data changes from Drupal`)
+        setPluginStatus({ lastFetched: data.data.timestamp })
+        requireFullRebuild = true
+      } else {
+        // Touch nodes so they are not garbage collected by Gatsby.
+        const existingNodes = getNodes().filter(
+          n => n.internal.owner === `gatsby-source-drupal`
+        )
+        existingNodes.forEach(n => touchNode({ nodeId: n.id }))
+
+        // Process sync data from Drupal.
+        let nodesToSync = data.data.entities
+        for (const nodeSyncData of nodesToSync) {
+          if (nodeSyncData.action === `delete`) {
+            actions.deleteNode({ node: getNode(createNodeId(nodeSyncData.id)) })
+          } else {
+            // The data could be a single Drupal entity or an array of Drupal
+            // entities to update.
+            let nodesToUpdate = nodeSyncData.data
+            if (!Array.isArray(nodeSyncData.data)) {
+              nodesToUpdate = [nodeSyncData.data]
+            }
+
+            for (const nodeToUpdate of nodesToUpdate) {
+              await handleWebhookUpdate(
+                {
+                  nodeToUpdate,
+                  actions,
+                  cache,
+                  createNodeId,
+                  createContentDigest,
+                  getCache,
+                  getNode,
+                  reporter,
+                  store,
+                },
+                pluginOptions
+              )
+            }
+          }
+        }
+
+        setPluginStatus({ lastFetched: data.data.timestamp })
+      }
+    } catch (e) {
+      gracefullyRethrow(drupalFetchIncrementalActivity, e)
```

Comment:
Should this `drupalFetchIncrementalActivity.end()` too?

Reply:
The activity is ended in the [gracefullyRethrow function](https://github.com/gatsbyjs/gatsby/blob/caea9adff18c169b1521109da653acf25c87dfd1/packages/gatsby-source-drupal/src/gatsby-node.js#L18)